### DISCUSSION TITLE
How to set a velocity inlet boundary condition without specifying a temperature on the boundary.

### DISCUSSION THREAD

**Discussion by akshaymani94** [nextfoam/baram#133] [Q&A]
I am facing a problem wherein I want my inlet boundary patch to allow entry of air into the domain at a particular velocity. But I don't want to specify the temperature of the entering air, I want it to be calculated based on the flow conditions around it. The velocity inlet boundary condition requires me to explicitly specify a temperature at the inlet. 
![Screenshot from 2024-10-22 18-20-09](https://github.com/user-attachments/assets/6f1f4895-6746-4fa1-a936-005e6e2667cf)
Please help me solve this issue.

**Comment by kdg5424** [comment]
Hello @akshaymani94. This is Donggyu from NEXTFOAM.

I guess you turn on and include 'Energy Model'.
You can change it 'Setup - Models - Energy Model'. Just off Energy Model.

Best regards

Donggyu Kim

**Reply by akshaymani94** [reply]
Ok. I dont want to confuse you. In my simulation air is entering at a temperature of 323K at 10m/s  and leaving through one outlet patch at 6m/s and the remaining air is leaving through a pressure outlet boundary condition. I have specified this outlet patch as velocity inlet and specified a negative velocity. I don't want to specify a temperature on this patch. The aim of this simulation is to find out the temperature of air leaving through this patch. Does this make sense.

**Reply by akshaymani94** [reply]
I have one more followup question.
How do I set the operating temperature. Say, I want the domain to be at 310K. I tried initializing with 310K as value of temperature, but it didn't work, it took a default value of 300K. I felt the requirement of this especially when I set my inlet as ABL inlet boundary condition where I don't get an option to specify the temperature of the entering air.

**Reply by kdg5424** [reply]
Hello @akshaymani94. This is Donggyu from NEXTFOAM.

1) Well, I got what you intended. I think you want to see the result like the below picture.
![image](https://github.com/user-attachments/assets/e5bdf851-60c8-4d91-ad2b-543392d18512)
I suggest you why don't you just configure as two "pressure outlet".

If you specify the velocity at one outlet, you should also specify correct temperature at there to get the right and precise result. If not, the result would be wrong. Because the temperature would be calculated by the energy equation and there is a term about kinetic energy. So velocity and temperature at outlet wolud affect each other and also affect the other outlet.
It means you should specify the right temperature and velocity at the outlet.

But I guess you don't know at the temperautre at the outlet then I suggest you just to configure as pressure outlet at both outlet.
Unfortunately, I'm not sure it would casuse the right result but it will be better than your current setup.

2) ABL with energy equation is not available in current version. There are some issue with thermal buoyancy effect with ABL inlet B.C.
We're developing ABL inlet with energy equation, so please wait until then.

Thanks

Best regards

Donggyu Kim

**Reply by akshaymani94** [reply]
![image](https://github.com/user-attachments/assets/fce7d75b-31b5-4ce4-8bba-2f9405f116c9)
The above shown is the schematic of my problem in 2D. I cannot configure it as a pressure outlet because the air is getting sucked out of the domain at a particular velocity (at location 1 in the image) which is known. Heated air is entering into the domain at 2. I want to find out whether some of this air is exiting through 1. For this I want to check the temperature at 1.

Thankyou for the information regarding the ABL.

**Reply by kdg5424** [reply]
Ah I know what you mean although it's not 100%. I guess you want to calculate the temperature at the inlet (And it's near 1 right?).
Well it's so unclear.
Could you give your stl or mesh?
I'll just look around

Best regards

**Comment by kdg5424** [comment]
Hello @akshaymani94. This is Donggyu from NEXTFOAM. I discussed with our engineers for your case.

And we drew a conclusion that the boundary condition what you want is not included in BARAM yet.
I guess we need to implement a boundary condition like exhaust fan in ANSYS Fluent. We'll add it in our roadmap. If you fund us, we can develop it with high priority.

Thanks

Donggyu Kim
